src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,39 @@
 package core.basesyntax.service;
 
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
+import java.util.Objects;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int MIN_AGE = 18;
+    public static final int MIN_PASSWORD_CHARACTERS_AMOUNT = 6;
 
     @Override
     public User register(User user) {
-        return null;
+        StorageDao storageDao = new StorageDaoImpl();
+        User userFromDb = storageDao.get(user.getLogin());
+
+        if (user.getAge() == 0
+                || user.getLogin() == null
+                || user.getPassword() == null
+                || user.getLogin().isEmpty()
+                || user.getPassword().isEmpty()) {
+            throw new NullPointerException("This line cannot be empty");
+        }
+        for (User person : Storage.people) {
+            if (Objects.equals(user.getLogin(), person.getLogin())) {
+                throw new RuntimeException(
+                        "User with this login " + user.getLogin() + " is already exists");
+            }
+            if (userFromDb.getAge() <= MIN_AGE) {
+                throw new RuntimeException("User's age is under 18");
+            }
+            if (userFromDb.getPassword().length() <= MIN_PASSWORD_CHARACTERS_AMOUNT) {
+                throw new RuntimeException("Password contains less then 6 characters");
+            }
+        }
+        return storageDao.add(userFromDb);
```

Comment:
```suggestion
    if (user == null
            || user.getAge() == null
            || user.getLogin() == null
            || user.getPassword() == null
            || user.getLogin().isEmpty()
            || user.getPassword().isEmpty()) {
      throw new NullPointerException("This line cannot be empty");
    }
    User userFromDb = storageDao.get(user.getLogin());

    if (userFromDb != null) {
      throw new RuntimeException(
              "User with this login " + user.getLogin() + " is already exists");
    }

    if (user.getAge() < MIN_AGE) {
      throw new RuntimeException("User's age is under 18");
    }

    if (user.getPassword().length() < MIN_PASSWORD_CHARACTERS_AMOUNT) {
      throw new RuntimeException("Password contains less then 6 characters");
    }

    return storageDao.add(user);
```

Тебе нужно проводить валидацию полей экзампляра класса User, который приходит тебе на вход в метод `public User register(User user)` а не того, которого ты хочешь достать из базы. Если у нас уже есть пользователь в базе с таким же логином, мы не должны регисторировать нового пользователя и выбросить ошибку